src/main/java/core/basesyntax/processdata/ProcessDataImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.processdata;
+
+import core.basesyntax.db.StorageDao;
+import core.basesyntax.db.StorageDaoImpl;
+import core.basesyntax.operationstrategy.OperationStrategy;
+import core.basesyntax.operationstrategy.OperationStrategyImpl;
+import core.basesyntax.operationstrategy.operation.OperationHandler;
+import java.util.List;
+import java.util.Map;
+
+public class ProcessDataImpl implements ProcessData {
+    private static final int OPERATION_TYPE_POSITION = 0;
+    private static final int ITEM_NAME_POSITION = 1;
+    private static final int ITEM_VALUE_POSITION = 2;
+    private final OperationStrategy operationStrategy;
+    private final StorageDao storageDao;
+
+    public ProcessDataImpl(Map<Character, OperationHandler> operationHandlerMap) {
+        operationStrategy = new OperationStrategyImpl(operationHandlerMap);
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Override
+    public Map<String, Integer> process(List<List<String>> processedData) {
+        for (List<String> dataLine : processedData) {
+            Character operationType = dataLine.get(OPERATION_TYPE_POSITION).charAt(0);

```

Comment:
```suggestion
            Character operationType = dataLine.get(OPERATION_TYPE_POSITION);
```
It’s hardcode think how to cast it in another way. Also,  I recommend you create enum Operation (see example  in Readme)